app/src/main/java/com/google/android/gms/samples/vision/ocrreader/aadhaar/utils/Storage.java

```diff
@@ -0,0 +1,92 @@
+package com.google.android.gms.samples.vision.ocrreader.aadhaar.utils;
+
+import android.content.Context;
+import android.util.Log;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+
+/**
+ * Created by yoti on 24/11/2016.
+ */
+
+public class Storage {
+    protected Context mContext;
+    protected static final String STORAGE_FILE_NAME = "data_storage.txt";
+
+    /**
+     * constructor
+     * @param activity
+     */
+    public Storage(Context activity){
+        mContext = activity;
+    }
+
+    /**
+     * Write to storage file
+     * @param data
+     */
+    public void writeToFile(String data) {
+        try {
+            OutputStreamWriter outputStreamWriter = new OutputStreamWriter(mContext.openFileOutput(STORAGE_FILE_NAME, Context.MODE_PRIVATE));
+            outputStreamWriter.write(data);
+            outputStreamWriter.close();
+        }
+        catch (IOException e) {
+            Log.e("Exception", "File write failed: " + e.toString());
+        }
+    }
+
+    /**
+     * Read from storage file
+     * @return String
+     */
+    public String readFromFile() {
+        // ensure file is created
+        checkFilePresent(STORAGE_FILE_NAME);
+
+        String ret = "";
+
+        try {
+            InputStream inputStream = mContext.openFileInput(STORAGE_FILE_NAME);
+
+            if ( inputStream != null ) {
+                InputStreamReader inputStreamReader = new InputStreamReader(inputStream);
+                BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
+                String receiveString = "";
+                StringBuilder stringBuilder = new StringBuilder();
+
+                while ( (receiveString = bufferedReader.readLine()) != null ) {
+                    stringBuilder.append(receiveString);
+                }
+
+                inputStream.close();
+                ret = stringBuilder.toString();
+            }
+        }
+        catch (FileNotFoundException e) {
+            Log.e("Storage activity", "File not found: " + e.toString());
+        } catch (IOException e) {
+            Log.e("Storage activity", "Can not read file: " + e.toString());
+        }
+
+        return ret;
+    }
+
+    private void checkFilePresent(String fileName) {
```

Comment:
This should be renamed as it does only check if the file is present it also creates it if it not there